### ISSUE TITLE
Backspace in hanging process?

### ISSUE THREAD

**Issue by tmpm697** [gnotclub/xst#100]
xst 1:0.8.4.r0.gda50afd-1
os: archlinux up-to-date
step:

```
> sleep 1000000
fdafdsfsfdasf^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^C
```
`^H` is backspace, I expect it will del previous char as in normal when edit command in prompt, I don't have issue with previous xst.

**Comment by actionless** [comment]
sidenote: in latest `st` this problem also not reproducible

**Comment by actionless** [comment]
@neeasade 

in xst config.def.h i see:

```c
	{ XK_BackSpace,     Mod1Mask,       "\033\177",      0,    0},
```

while in st i see:

```c
	{ XK_BackSpace,     XK_NO_MOD,      "\177",          0,    0},
	{ XK_BackSpace,     Mod1Mask,       "\033\177",      0,    0},
```

**Comment by tmpm697** [comment]
yes, adding `{ XK_BackSpace,     XK_NO_MOD,      "\177",          0,    0},` worked.

**Comment by actionless** [comment]
@neeasade should we fix it in xst?

**Comment by neeasade** [comment]
@actionless yes, will PR

**Comment by kevindirect** [comment]
I don't know if this is related to my problem but I notice now that I have to backspace twice to delete the first character in a string now. Subsequent backspaces work properly, but the first one requires to backspace presses.

**Comment by tmpm697** [comment]
You might increase the keyrate in X window -- .xserverrc?

**Comment by kevindirect** [comment]
I already decrease the xset rate delay for all keys in .xinitrc. I can try adding an extra delay for backspace. I also noticed the standard st package from the AUR doesn't have this issue. I think this started with a recent update of xst. Thanks for the suggestion though I'm going to try it if I don't find another resolution.

**Comment by actionless** [comment]
> this started with a recent update of xst

could you please do git-bisect-ing for your usecase?

**Comment by kevindirect** [comment]
I started at git tag v0.8.4. I got this output:

> 079cf8cd6bc811056750a10ec876d23a50ea6cce is the first bad commit
> commit 079cf8cd6bc811056750a10ec876d23a50ea6cce
> Author: neeasade <neeasade@users.noreply.github.com>
> Date:   Sun Jul 12 16:44:28 2020 -0500
> 
>     backspace definition -- fixes https://github.com/gnotclub/xst/issues/100 (#105)
> 
>  config.def.h | 1 +
>  1 file changed, 1 insertion(+)
>

**Comment by actionless** [comment]
@kevindirect 

any difference when applying this patch? https://github.com/gnotclub/xst/pull/107/files

**Comment by actionless** [comment]
sorry, the new link to PR: https://github.com/gnotclub/xst/pull/109 (for some reason force-pushing to closed PR makes it not possible to re-open it back)

**Comment by actionless** [comment]
(in order to test that patch you'll need either to apply locally #108 or re-generate terminfo-s with `tic` manually)

**Comment by neeasade** [comment]
this issue is resolved for me with #109

**Comment by kevindirect** [comment]
Same, issue was resolved for me on revision 11. Thanks
